doc/dev/references/testing/integration-testing/index.md

```diff
@@ -0,0 +1,32 @@
+> *[Testing](../index.md) > Integration testing*
+>
+> See also: [How to write a integration test](/doc/dev/how-to/write-an-integration-test.md)
+
+[test folder]: /tests
+
+[test suites]: /tests/suites
+
+[test includes]: /tests/includes
+
```

Comment:
These link definitions are confusing here at the top. I'd move them to the end or in-line.

Reply:
They won't be rendered. I put it on top because it's easier to maintain. But I can move them to the end (we just need to decide where is the proper place). However i don't like inline link it make reading source markdown painful and is error-prone AMHO.

Reply:
> I put it on top because it's easier to maintain. 

Good point. I do think we should follow the same convention everywhere. From past experience I think in the Juju codebase they tend to be at the end. (I could be wrong.) Basically, whatever the current convention is, let's follow that.

Reply:
That's used only in doc/dev in the current repository:

![image](https://github.com/user-attachments/assets/e5101989-27e3-4b1a-943e-da10d6f6ec12)

So i introduce the pattern, following the following rule of thumb:

"Put it at the beginning of the scope where the keyword with a link is relevant"

However A good test would be to check if a same keyword could lead to a different URL regarding section. Let's test

```md
## Google

[Search Engine]: https://google.com/

Please use the [Search Engine]

## Duck duck go

[Search Engine]: https://duckduckgo.com/

Please use the [Search Engine]
```

## Google

[Search Engine]: https://google.com/

Please use the [Search Engine]

## Duck duck go

[Search Engine]: https://duckduckgo.com/

Please use the [Search Engine]


==> As we can see "the first defined, the winner".

So I think if we use such thing we should put them at the beginning OR the end of a file, and follow the same convention consistently.

For me both have pros and cons.

* Beginning: easier to not forget and to get the convention. But cause clutter for people who read from source instead of rendered file
* End: the exact opposite :)

Reply:
Thank you for the careful discussion. In the codebase I've noticed they're kept at the end. Maybe because there people read from source. For these higher-level docs we anticipate that people might read them from the rendered files. As such, by your logic, here we should put them at the beginning. However, there's always a cost to having a split standard (people get confused and you end up having no standard at all). I say let's keep them at the end in both cases. 

PS Personally I prefer in-line for all.